src/components/NewTaxonomy/NewTaxonomy.tsx

```diff
@@ -0,0 +1,98 @@
+import { TreeSelect } from 'antd';
+import React, { useCallback, useEffect, useState } from 'react';
+
+type TaxonomyPath = string[];
+type onAddLabelCallback = (path: string[]) => any;
+type onDeleteLabelCallback = (path: string[]) => any;
+
+type TaxonomyItem = {
+  label: string,
+  path: TaxonomyPath,
+  depth: number,
+  children?: TaxonomyItem[],
+  origin?: 'config' | 'user' | 'session',
+  hint?: string,
+};
+
+type AntTaxonomyItem = {
+  title: string,
+  value: string,
+  key: string,
+  children?: AntTaxonomyItem[],
+};
+
+type TaxonomyOptions = {
+  leafsOnly?: boolean,
+  showFullPath?: boolean,
+  pathSeparator: string,
+  maxUsages?: number,
+  maxWidth?: number,
+  minWidth?: number,
+  placeholder?: string,
+};
+
+type TaxonomyProps = {
+  items: TaxonomyItem[],
+  selected: TaxonomyPath[],
+  onChange: (node: any, selected: TaxonomyPath[]) => any,
+  onLoadData?: (item: TaxonomyPath) => any,
+  onAddLabel?: onAddLabelCallback,
+  onDeleteLabel?: onDeleteLabelCallback,
+  options?: TaxonomyOptions,
+  isEditable?: boolean,
+};
+
+const convert = (items: TaxonomyItem[], separator: string): AntTaxonomyItem[] => {
+  return items.map(item => ({
+    title: item.label,
+    value: item.path.join(separator),
+    key: item.path.join(separator),
+    // disableCheckbox: !!item.children,
+    isLeaf: item.isLeaf !== false && !item.children,
+    // checkable: !item.children,
+    // selectable: !item.children,
+    children: item.children ? convert(item.children, separator) : undefined,
```

Comment:
are the commented out lines intended to be added in later? or can we remove them?